src/game.cpp

```diff
@@ -319,11 +320,18 @@ void game::load_core_data()
     // core data can be loaded only once and must be first
     // anyway.
     DynamicDataLoader::get_instance().unload_data();
+    MonsterGenerator::generator().reset(); // @todo Separate clearing from destroying. It's not the best place for keeping it.
```

Comment:
> Regarding the comment reset here resets the content, it does not destruct the generator.

And that's a good thing, because it's in `game::load_core_data()`, not in a destructor.

> reset should be called from unload_data there is no reason to move this out of there.

There [is](https://github.com/CleverRaven/Cataclysm-DDA/pull/17093/files#r66237782) a reason:

> With this line here, the game class tries to reset the already destroyed object.